app-editors/helix/helix-23.10.ebuild

```diff
@@ -279,6 +279,8 @@ SLOT="0"
 KEYWORDS="~amd64"
 IUSE="+grammar"
 
+BDEPEND="grammar? ( dev-vcs/git )"
```

Comment:
Am I understanding correctly that these grammars are fetched at runtime? Because they can't be fetched at buildtime due to the network sandbox. So it must RDEPS instead of BDEPS?

Reply:
https://github.com/helix-editor/helix/blob/992b7a0c395b5c43e2fd4f6eee38bed8dda9721a/helix-term/build.rs#L4
https://github.com/helix-editor/helix/blob/992b7a0c395b5c43e2fd4f6eee38bed8dda9721a/flake.nix#L130

Reply:
> Am I understanding correctly that these grammars are fetched at runtime when invoking hx? Because they can't be fetched at buildtime due to the network sandbox. So it must RDEPS instead of BDEPS?

No, they are (normally) fetched at build time. But when all the grammar sources are already present (which is the case with this ebuild) the fetch will be skipped and the build starts. But I think `git` is still needed to verify some things? They verify the existance of `git` not only in the fetch function, but also in the build function. The reasoning was described in the commit I posted.

> Ok, so the upstream commit you posted didnt have this, but the build time fetching can be disabled as nix does it already.

We use the same mechanism to enable/disable the grammar build:
https://github.com/gentoo/gentoo/blob/master/app-editors/helix/helix-23.10.ebuild#L295